src/NServiceBus.Core/Performance/Metrics/MetricsExtensions.cs

```diff
@@ -0,0 +1,28 @@
+namespace NServiceBus
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Extension methods to configure OpenTelemetry metrics.
+    /// </summary>
+    public static class MetricsExtensions
+    {
+        /// <summary>
+        /// Configures NServiceBus to report messaging metrics using OpenTelemetry.
+        /// </summary>
+        /// <param name="config">The <see cref="EndpointConfiguration" /> instance to apply the settings to.</param>
+        public static void EnableOpenTelemetryMetrics(this EndpointConfiguration config)
+        {
+            Guard.AgainstNull(nameof(config), config);
+            config.EnableFeature<MessagingMetricsFeature>();
+        }
+    }
+
+    static class Extensions
+    {
+        internal static bool TryGetMessageType(this IReadOnlyDictionary<string, string> headers, out string processedMessageTypes)
```

Comment:
Cosnider moving this into `ReceiveDiagnosticsBehavior` as that's the only place it is called from at the moment.

Reply:
Moved